d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerTest.java

```diff
@@ -1475,16 +1475,12 @@ else if (partitionMethod == 1)
           assertEquals(unmappedKeys.size(), 1);
         }
 
-        try
-        {
-          loadBalancer.getClient(new URIRequest("d2://foo/id=100"), new RequestContext());
-          if (partitionMethod == 0)
-          {
-            // key out of range
-            fail("Should throw ServiceUnavailableException caused by PartitionAccessException");
-          }
+        loadBalancer.getClient(new URIRequest("d2://foo/id=100"), new RequestContext());
+        if (partitionMethod == 0) {
+          // key out of range
+          assertEquals(mapKeyResult.getUnmappedKeys().size(), 1);
```

Comment:
Shouldn't we test that loadBalancer.getClient is returning something in the default partition? The mapKeyResult is from previous test scenario mapKeyResult = loadBalancer.getRings(URI.create("d2://foo"), keys);